polaris-react/src/components/Button/Button.tsx

```diff
@@ -288,7 +300,12 @@ export function Button({
   };
 
   const buttonMarkup = (
-    <UnstyledButton {...commonProps} {...linkProps} {...actionProps}>
+    <UnstyledButton
+      {...commonProps}
+      {...linkProps}
+      {...actionProps}
+      preventedInteraction={preventedInteraction}
```

Comment:
Similar to above, we should be handling the nullifying of the event within the existing handlers within the UnstyledButton component, so we don't need to pass this prop through.